### ISSUE TITLE
Obtain probabilities mask

### ISSUE THREAD

**Issue by lcerdaal** [MIC-DKFZ/nnUNet#1032]
Hello!

I would like to obtain the predicted probabilities per voxel (of belonging to class 0 or 1) instead of the default segmentation mask with 0 / 1 values. Could you please help me understand where I should apply this modification?

Also, is there any output file where I can see the parameters that have been selected for model training? For instance, I have read that the initial batch size is 2 (3D model), but where can I see if this parameter has been optimized for my particular dataset and what is its final value?

Apologies if these are too obvious questions, and thank you very much for your help!

Best,
Leo

**Comment by julien12234** [comment]
Hello! I would also be interested in the answer to the question about the probabilities per voxel. My model seems to undersample the volume, I would like to see if I can change the threshold? Or is this parameter not fixed but optimized? 

Best, 
Julien

**Comment by daidadwawdfa** [comment]
U can get it by set 'save npz' But the softmax size might not be the same as the GT mask size

**Comment by FabianIsensee** [comment]
There is no threshold since we use argmax (!). So in a 2-class case one could argue that the threshold is 0.5. In multi-class segmentation there is no clear threshold here...

save_npz will resample the softmax to the spacing of the input images but it will not revert the cropping. This you will need to do yourself (see https://github.com/MIC-DKFZ/nnUNet/blob/d0ba941a8a8996dca3ac36afc34988bafbc575a1/nnunet/inference/segmentation_export.py#L27)
